### PR TITLE
new style inv and batch_inv function

### PR DESCRIPTION
**Pull request by kmaehashi**
This PR implements new style `inv` and `batch_inv` function.

### CODE REVIEW

**Comment by takagi** [comment]
Travis fails because of a problem fixed in #3599. Would you rebase the master?

**Comment by kmaehashi** [comment]
Rebased.

**Comment by takagi** [comment]
jenkins, test this please.

File: chainer/functions/math/inv.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import numpy.linalg
 
 from chainer import cuda
-from chainer import function
+from chainer import function_node
+from chainer.functions.array import transpose
```
Comment (takagi) [inline chainer/functions/math/inv.py]: Please use `chaner.functions.transpose`.

File: chainer/functions/math/inv.py [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 from chainer.functions.math import matmul
```
Comment (takagi) [inline chainer/functions/math/inv.py]: Please use `chainer.functions.matmul`.

**Comment by takagi** [comment]
jenkins, test this please.

**Comment by kmaehashi** [comment]
Thanks for reviewing! Fixed accordingly.
Note that this module uses private functions from `matmul.py` so `from chainer.functions.math import matmul ` cannot be removed.
e.g., https://github.com/chainer/chainer/pull/3616/files#diff-2e2780f463ffaee21f04cc17516d6bc4R15

**Comment by takagi** [comment]
jenkins, test this please.

**Comment by takagi** [comment]
LGTM!

**Comment by takagi** [comment]
@kmaehashi Would you add double backward test for the new `inv` and `batch_inv` functions? I'm sorry that I missed to find it on my review.

**Pull request merged**